src/xc_integrator/xc_data/device/xc_device_stack_data.cxx

```diff
@@ -273,7 +289,30 @@ void XCDeviceStackData::send_static_data_density_basis( const double* P, int32_t
   if( not device_backend_ ) GAUXC_GENERIC_EXCEPTION("Invalid Device Backend");
 
   // Copy Density
-  device_backend_->copy_async( nbf*nbf, P, static_stack.dmat_device, "P H2D" );
+  device_backend_->copy_async( nbf*nbf, P, static_stack.dmat_s_device, "P H2D" );
+
+  // Copy Basis Set
+  device_backend_->copy_async( basis.nshells(), basis.data(), static_stack.shells_device,
+    "Shells H2D" );
+
+
+  device_backend_->master_queue_synchronize(); 
+}
+
+void XCDeviceStackData::send_static_data_density_basis( const double* Ps, int32_t ldps, const double* Pz, int32_t ldpz, 
```

Comment:
To avoid copying code - you should just call the "RKS" function overload within this function and simply extend it with the UKS data.